### PR TITLE
Bring back handing the fighting style feat addition.

### PR DESCRIPTION
**Pull request by ThyWoof**

### CODE REVIEW

File: SolastaCommunityExpansion/Models/CustomFeaturesContext.cs [hunk 1]
```diff
@@ -19,6 +19,26 @@ internal static void RecursiveGrantCustomFeatures(RulesetCharacterHero hero, Lis
...
+                if (!(grantedFeature is FeatureDefinitionProficiency featureDefinitionProficiency))
+                {
+                    continue;
+                }
+
+                if (featureDefinitionProficiency.ProficiencyType != RuleDefinitions.ProficiencyType.FightingStyle)
+                {
+                    continue;
+                }
+                using (var enumerator = featureDefinitionProficiency.Proficiencies.GetEnumerator())
```
Comment (PhilPJL) [inline SolastaCommunityExpansion/Models/CustomFeaturesContext.cs:32]: This should probably be

```
foreach(var proficiency in featureDefinitionProficiency.Proficiencies)
{
    var definition = DatabaseRepository.GetDatabase<FightingStyleDefinition>().GetElement(key, false);

    if(definition == null)
    {
        throw new SolastaModApiException("... ");
    } 
hero.TrainedFightingStyles.Add(definition);
```

    
}

Comment (PhilPJL) [reply SolastaCommunityExpansion/Models/CustomFeaturesContext.cs:32]: GetElement does this which really isn't very helpful if the key doesn't exist

```
    public T GetElement(string key, bool allowFail = false)
    {
        T value = null;
        if (valuesTable.TryGetValue(key, out value))
        {
            return value;
        }

        if (!allowFail)
        {
            Trace.LogError("Cannot find element {0} in database {1}", key, Name);
        }

        return value;
    }
```

**Review by PhilPJL — CHANGES_REQUESTED** [review]
See comment

File: SolastaCommunityExpansion/Models/CustomFeaturesContext.cs (hunk 1 above)
Comment (ChrisPJohn) [reply SolastaCommunityExpansion/Models/CustomFeaturesContext.cs:32]: I agree a for loop would be better. I was simply bringing back code that was wrongly deleted in a refactor.

**Comment by ChrisPJohn** [comment]
I didn't make this pull request I'm closing it.

**Pull request closed without merging**